ibridges/interactive.py

```diff
@@ -39,10 +39,13 @@ def interactive_auth(password: Optional[str] = None,
         print(f'File not found: {irods_env_path}')
         raise FileNotFoundError
 
-    if os.path.exists(Path(os.path.expanduser("~")).joinpath(".irods", ".irodsA")):
+    if os.path.exists(Path(os.path.expanduser("~")).joinpath(".irods", ".irodsA")) and \
+            password is None:
         try:
             session = Session(irods_env_path)
             return session
+        except LoginError:
```

Comment:
Can you explain this bit of code? It seems like it has no effect.

Reply:
This one changed quite a bit.

```py
 47         except LoginError as e:
 48             print(f'INFO: {e.args}')
 49             # environment.json is not setup correctly -> stop
 50             if 'not set correctly in irods_environment.json' in e.args[0]:
 51                 raise(e)
 52             # non-iRODS key found
 53             if 'Unexpected value in irods_environment; ' in e.args:
 54                 raise(ValueError('Non-iRODS key or value found in environment.json')) from     e
```

The later versions of the PRC (1.1.9 and 2.0.0) always throw an error when they encounter something strange in the environment.json. So the user needs to stop and fix that first.
Unfortunately, the PRC 2.0.0 does not make use of the previous Exceptions but always throws a:

```
ValueError("Authentication failed: scheme = 'pam', auth_type = None")
```

Hence the not so straight forward solution here.